app/views/returns/submittedReturns.scala.html

```diff
@@ -76,7 +76,11 @@ <h2>@messages("submittedReturns.yearReturns", model.selectedYear.toString)</h2>
                 <a href="@controllers.routes.ReturnsController.vatReturn(model.selectedYear, obligation.periodKey)"
                    data-journey-click="returns:view-return-@{obligation.start}-to-@{obligation.end}:submitted-returns">
                   <span class="visually-hidden">@messages("submittedReturns.periodPrefix")</span>
-                  @displayDateRange(obligation.start, obligation.end)
+                  @if(obligation.periodKey==appConfig.finalReturnPeriodKey) {
```

Comment:
Can you create a def on the model called `isFinalObligation` that as the appConfig passed implicitly to it. That way, the view logic can be simplified to `@if(isFinalObligation)` - also, it means the method can be reused elsewhere.